fgpyo/sam/__init__.py

```diff
@@ -606,6 +608,118 @@ def __getitem__(
         return self.elements[index]
 
 
+@enum.unique
+class PairOrientation(enum.Enum):
+    """Enumerations of read pair orientations."""
+
+    FR = "FR"
+    """A pair orientation for forward-reverse reads ("innie")."""
+
+    RF = "RF"
+    """A pair orientation for reverse-forward reads ("outie")."""
+
+    TANDEM = "TANDEM"
+    """A pair orientation for tandem (forward-forward or reverse-reverse) reads."""
+
+    @classmethod
+    def from_recs(  # noqa: C901  # `from_recs` is too complex (11 > 10)
```

Comment:
I couldn't really think of a way of simplifying this function without making harder to grok in isolation private subfunctions... and 11 vs 10 didn't seem so bad.

Reply:
Honestly, I don't like the complexity check.  Sometimes things _are_ complex, and breaking it out into functions arbitrarily just makes it harder to follow what's going onl